### PR TITLE
fix: allow paginating queries that return non scalars - DIA-29942

### PR DESCRIPTION
**Pull request by hadrien**
## Description

* Allow paginating queries that do not return scalars;

### CODE REVIEW

**Comment by hadrien** [comment]
@vicrep As you know, I am not fond of using booleans, but, I do not have a good understanding on all types of query it would have to handle ... so good enough for now.

File: README.md [hunk 1]
```diff
@@ -177,6 +180,46 @@ By default:
     }
     ```
 
+#### Paginating non-scalar results
+
+When query doesn't returned scalars, specify `scalars=False` when invoking `paginate`:
```
Comment (vicrep) [inline README.md:185]:
```suggestion
To paginate a query which doesn't return scalars, specify `scalars=False` when invoking `paginate`:
```

Also, is scalar a known/common enough concept, or should we perhaps link to associated docs here?

File: fastapi_sqla/__init__.py [hunk 2]
```diff
@@ -243,13 +245,16 @@ def _paginate(
     total_items: int,
     offset: int,
     limit: int,
+    *,
+    scalars: bool = True,
```
Comment (vicrep) [inline fastapi_sqla/__init__.py:249]: Should this be singular `scalar`? (I'm not sure, I'm asking you @hadrien )

Comment (vicrep) [reply fastapi_sqla/__init__.py:249]: JK, I got confused between the old `.scalar()` from the legacy queries and `Results.scalars()`

**Review by vicrep — APPROVED** [review]

File: fastapi_sqla/__init__.py (hunk 2 above)
Comment (hadrien) [reply fastapi_sqla/__init__.py:249]: it is plural as it return multiple scalar.

**Pull request merged**